File: 1. cpp/1. basics/lecture 2/p12.cpp

#include <iostream>
using namespace std;
class solution {
public:
    solution(int n) {
        for (int i = 1; i <= n; i++) {
            // first half
            for (int j = 1; j <= i; j++) {
                cout << j << " ";
            }
            // spaces
            int spaces = 2 * (n - i);
            for (int j = 1; j <= spaces; j++) {
                cout << "  ";
            }
            // second half
            for (int j = i; j >= 1; j--) {
                cout << j << " ";
            }
            cout << endl;
        }
    }
};

/* pattern output:

1         1
1 2     2 1
1 2 3 3 2 1

*/

int main() {
  int size;
  cout<<"Enter size of pattern:"<<endl;
  cin>>size;

  cout<<"7. pyramid :"<<endl;
  solution pattern(size);

}

File: 2. python/1. basics/lecture 2/p12.py
class Solution:
    def __init__(self, n):
        for i in range(1, n + 1):
            # first half
            for j in range(1, i + 1):
                print(j, end=' ')
            # spaces
            for j in range(1, 2 * (n - i) + 1):
                print(' ', end=' ')
            # second half
            for j in range(i, 0, -1):
                print(j, end=' ')
            print()

''' pattern output:

1         1
1 2     2 1
1 2 3 3 2 1

'''

if __name__ == "__main__":
    size = int(input("enter size: "))

    print("1. output:")
    Solution(size)